nerfstudio/data/pixel_samplers.py

```diff
@@ -92,3 +92,66 @@ def sample(self, image_batch: Dict):
             image_batch, self.num_rays_per_batch, keep_full_image=self.keep_full_image
         )
         return pixel_batch
+
+
+def collate_image_dataset_batch_equirectangular(batch: Dict, num_rays_per_batch: int, keep_full_image: bool = False):
+    """
+    Operates on a batch of equirectangular images and samples pixels to use for
+    generating rays. Rays will be generated uniformly on the sphere.
+    Returns a collated batch which is input to the Graph.
+    It will sample only within the valid 'mask' if it's specified.
+
+    Args:
+        batch: batch of images to sample from
+        num_rays_per_batch: number of rays to sample per batch
+        keep_full_image: whether or not to include a reference to the full image in returned batch
+    """
+    # TODO(kevinddchen): make more DRY
+    device = batch["image"].device
+    num_images, image_height, image_width, _ = batch["image"].shape
+
+    # only sample within the mask, if the mask is in the batch
+    if "mask" in batch:
+        # TODO(kevinddchen): implement this
+        raise NotImplementedError("Masking not implemented for equirectangular images.")
+    else:
+        num_images_rand = torch.rand(num_rays_per_batch, device=device)
+        # sample phis according to pdf(x) = sin(x) / pi, but normalized to the range [0, 1]
+        phi_rand = torch.acos(1 - 2 * torch.rand(num_rays_per_batch, device=device)) / torch.pi
```

Comment:
would it maybe be more intuitive to write this as calculate a random direction uniform over a unit sphere, then convert this to a pixel in the image? Can't seem to find where this math was found from

Reply:
I agree its doing something less intuitive.

Just making sure I'm thinking about this right: sampling uniformly over the sphere means picking `phi` and `theta`. `theta` is uniformly distributed over `[0, 2 pi]`, no confusion there. `phi` is distributed over `[0, pi]` with weight `sin(phi)`, since that is the differential area element. So the PDF is proportional to `sin(phi)`.

Here we are doing something strange, since we are using inverse transform sampling to draw random samples using samples drawn from the uniform distribution over [0, 1].

Would you like this documented better? Or is there a better way and I'm overcomplicating?

Reply:
Sampling over the unit sphere uniformly can be done by normallizing a vector made by 3 samples from a standard normal distribution:

> xyz = torch.normal(mean=torch.zeros((3,)))
> sample_xyz = xyz / xyz.norm()

It might be good to add documentation on similar to how we are doing here with the other camera types: https://docs.nerf.studio/en/latest/nerfology/model_components/visualize_cameras.html#

I haven't touched this part of the docs yet, but from what I understand its all notebook based so modifying one of the other graphics shouldn't be too different-- but let me know if I can help with that!

Reply:
Hmmm yeah, thats much more intuitive. I'll use that instead then.

I'll take a look at the docs and see if I understand what is going on there, haha.

Reply:
Hmmmm ok, going back to this, I did some thinking and while the 3d gaussian approach is more intuitive, its a lot harder to convert a point on the unit sphere back to coordinates in the equirectangular image.

To figure out the azimuthal angle, we will need some kind of `atan2(y, x)` but also condition on the signs of `x` and `y` and adding/subtracting multiples of `pi` to get things into the correct range.

I'm confident the current implementation is correct, so if you don't mind, can we go with this then? Its a lot simpler implementation-wise, even if the approach is less intuitive.

Reply:
@kevinddchen yup! That's totally good with me, just wanted to check the math was right and I hadn't seen the inverse transform sampling derivation before. I can also do a full review now and let you know if I see anything, but this all looks great so far. thanks again for the help 😀

Reply:
Yes please! Be super nit picky!

Reply:
You can reference this for a derivation for sampling using the inverse CDF (http://corysimon.github.io/articles/uniformdistn-on-sphere/)